docs/design.md

```diff
@@ -0,0 +1,144 @@
+# GCP Ingestion Design
+
+This document specifies the design for GCP Ingestion as a whole.
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+
+- [Architecture Diagram](#architecture-diagram)
+- [Data Flow](#data-flow)
+  - [Notes](#notes)
+- [Design Decisions](#design-decisions)
+  - [Kubernetes and PubSub](#kubernetes-and-pubsub)
+  - [Different topics for "raw" and "validated"](#different-topics-for-raw-and-validated)
+  - [BigQuery](#bigquery)
+  - [Saving as JSON](#saving-as-json)
+  - [Use destination tables?](#use-destination-tables)
+  - [Use views for user-facing data?](#use-views-for-user-facing-data)
+- [Known Issues](#known-issues)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Architecture Diagram
+
+![alt text](architecture.svg "Architecture Diagram")
+
+## Data Flow
+
+- The Kubernetes Edge service passes data from Producers (e.g. Firefox) to a
+  PubSub Raw Topic.
+   - The service will store data on disk when PubSub is unavailable.
+      - New messages always attempt delivery to PubSub first.
+      - Data on disk is retried periodically.
+      - The service cannot be autoscaled down when there is data on disk.
+   - The service will respond 500 if PubSub and disk space are both unavailable.
+      - This is to be avoided because it has numerous negative side-effects,
+        including delayed `submission_date` and usually some amount of data
+loss.
+   - The PubSub Topic for data is configurable based on `uri`.
+   - Expected topics are Structured Ingestion, Telemetry, and Pioneer.
+- The Dataflow Landfill Sink job copies data from a PubSub Raw Topic to Cloud
+  Storage.
+   - Data can be read from Cloud Storage for backfill, recovery, or testing.
+- The Dataflow Decoder job decodes data from a PubSub Raw Topic to a PubSub
+  Decoded Topic.
+   - Decoding data applies these transforms in order:
```

Comment:
Consider linking to the `decoder` subdirectory of `ingestion-beam` here since these steps map directly to the files there.